modules/core/src/controllers/orthographic-controller.js

```diff
@@ -105,15 +106,43 @@ class OrthographicState extends ViewState {
     return this;
   }
 
+  // Calculates new zoom
+  _calculateNewZoom({scale, startZoom}) {
+    const {maxZoom, minZoom} = this._viewportProps;
+    const zoom = startZoom + Math.log2(scale);
+    return clamp(zoom, minZoom, maxZoom);
+  }
+
+  /**
+   * Start zooming
+   * @param {[Number, Number]} pos - position on screen where the center is
+   */
+  zoomStart({pos}) {
+    return this._getUpdatedState({
+      startZoom: this._viewportProps.zoom
+    });
+  }
+
   /**
    * Zoom
    * @param {Number} scale - a number between [0, 1] specifying the accumulated
    *   relative scale.
    * @param {[number, number]} pos - current mouse cursor screen position
    */
   zoom({pos, scale}) {
-    const {zoom, width, height, offset, minZoom, maxZoom} = this._viewportProps;
-    const newZoom = clamp(zoom * scale, minZoom, maxZoom);
+    const {zoom, width, height, offset} = this._viewportProps;
+    let {startZoom} = this._interactiveState;
+    if (!Number.isFinite(startZoom)) {
+      // We have two modes of zoom:
+      // scroll zoom that are discrete events (transform from the current zoom level),
+      // and pinch zoom that are continuous events (transform from the zoom level when
+      // pinch started).
+      // If startZoom state is defined, then use the startZoom state;
+      // otherwise assume discrete zooming
+      startZoom = this._viewportProps.zoom;
+    }
+
+    const newZoom = this._calculateNewZoom({scale, startZoom});
     const centerX = width / 2 - offset[0];
     const centerY = height / 2 - offset[1];
     const dX = (pos[0] - centerX) * (zoom / newZoom - 1);
```

Comment:
Zoom center is broken. I believe this needs to be reversed as well.

Reply:
When you zoom in/out the plot is supposed to scale around your pointer location. On your branch:

![ortho-controller](https://user-images.githubusercontent.com/2059298/49410932-ccfb5080-f71b-11e8-84c7-a79cd73055b8.gif)

Reply:
![zoom](https://user-images.githubusercontent.com/3599601/49619256-b1d54e80-f970-11e8-95ac-fb8186bfc965.gif)